### DISCUSSION TITLE
GeoJSON feature properties supported by Obsidian Leaflet

### DISCUSSION THREAD

**Discussion by Moonbase59** [javalent/obsidian-leaflet#124] [Show and tell]
Here’s a table of **GeoJSON feature properties** that _Obsidian Leaflet_ currently supports. Valid as of 2021-06-20, version 3.18.14.

| Property         | Default        | Usage                                                                                                    |
| ---------------- | -------------- | -------------------------------------------------------------------------------------------------------- |
| `name`           | empty          | Tooltip title if no `title` property found.                                                              |
| `title`          | empty          | Tooltip title.                                                                                           |
| `description`    | empty          | If present, tooltip will show an ⓘ and, on click, show the title _and_ description.                      |
| `marker-symbol`  | default marker | Will display the named marker, if defined in plugin settings.                                            |
| `stroke`         | "#3388FF"      | The color of a line as part of a polygon, polyline, or multigeometry.                                    |
| `stroke-opacity` | 0.6            | The opacity of the line component of a polygon, polyline, or multigeometry.                              |
| `stroke-width`   | 2              | The width of the line component of a polygon, polyline, or multigeometry.                                |
| `fill`           | "#3388FF"      | The color of the interior of a polygon.                                                                  |
| `fill-opacity`   | 0.2            | The opacity of the interior of a polygon. Implementations may choose to set this to 0 for line features. |

A "Feature" (within a "FeatureCollection") in GeoJSON can look like this (example):

```json
    {
      "type": "Feature",
      "properties": {
        "title": "Jackson Road Extension",
        "description": "Extension of US 66 in 1937, upon completion of land reclamation for 1937 Chicago World's Fair",
        "stroke": "#ff0000",
        "stroke-width": 3
      },
      "geometry": {
        "type": "LineString",
        "coordinates": [
          [
            -87.617232,
            41.8784146,
            0
          ],
          [
            -87.6243238,
            41.8783107,
            0
          ]
        ]
      }
    }

```

**Comment by zhubib1234** [comment]
very nice!

**Comment by oyvindsolstad** [comment]
Is there a way to set things like "stroke-width" and "stroke-opacity" on a general level in a JSON-file? (may be a noob question). I have all US counties in a file and it seems like much work to set these for every county vs just setting it as a general value for the whole file.
